src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -683,6 +683,56 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
             return true; // to skip internal onPaste handler
         }
 
+        // Safari `Insert from iPhone or iPad`
+        // data.getData("text/html") returns a string like: <img src="blob:https://...">
+        if (data.types.includes("text/html")) {
+            const imgElementStr = data.getData("text/html");
+            const parser = new DOMParser();
+            const imgDoc = parser.parseFromString(imgElementStr, "text/html");
+
+            if (
+                imgDoc.querySelector("img") == null ||
+                imgDoc.querySelector("img")!.src == null ||
+                imgDoc.getElementsByTagName("img").length != 1 ||
+                !imgDoc.querySelector("img")!.src.startsWith("blob:")
+            ) {
+                console.log("Failed to handle pasted content as Safari inserted content");
+
+                // Fallback to internal onPaste handler
+                return false;
+            }
+            const imgSrc = imgDoc!.querySelector("img")!.src;
+
+            fetch(imgSrc).then(
+                (response) => {
+                    response.blob().then(
+                        (imgBlob) => {
+                            const type = imgBlob.type;
+                            const safetype = getBlobSafeMimeType(type);
+                            const ext = type.split("/")[1];
+                            const file = new File([imgBlob], "file." + ext, { type: safetype });
```

Comment:
Should this be internationalised?

Reply:
You mean `file` should be internationalized?

Well... that's okay but... Now if you paste an image it's `image.png` and it's not internationalized 🤣

Reply:
`image.png` comes from your browser, which may very well internationalise it

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/da33beab-ce5c-4b35-9dee-dcde0087be8b)

Reply:
Maybe we should just use the random blob identifier to form the name

Reply:
meaningful here just means every time someone downloads it they'll need to give it their own name or risk overwriting the previous file you pasted. Using the path part of the blob url also masks that you're on safari, indistinguishable from a random file name.

Reply:
Okay. Changed as required.